### PR TITLE
feat: add templates for debugging purposes

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR adds k8s templates for debug pods, i.e pods running with non-production setup (root user, container entrypoint/command changed, ...), which allow developers to debug services like LMS/CMS in a production-like environment.

### How to test
1. Configure debugging capabilities:
```
# config.yml
OPENEDX_DEBUG_COOKIE: ednx_enable_debug # or anything else
DRYDOCK_DEBUG: False
```
2. Generate new manifests
3. Sync in argoCD
4. Increase replicas for debug services
`k scale --replicas=1 deployment/lms-debug`
or
`k scale --replicas=1 deployment/cms-debug`
5. Go to the LMS/Studio page. Then, add your cookie:
`ednx_enable_debug: always`
6. Check the traffic is being redirected to your debugging pod.

You can test this in yamato using any namespace defined.

### CODE REVIEW

File: drydock/templates/kustomized/tutor14/extensions/hpa.yml [hunk 1]
```diff
@@ -78,3 +79,4 @@ spec:
     kind: Deployment
     name: cms-worker
   targetCPUUtilizationPercentage: {{ DRYDOCK_FORUM_TARGET_CPU }}
+{%- endif %}
```
Comment (mariajgrimaldi) [inline drydock/templates/kustomized/tutor14/extensions/hpa.yml:82]: This shouldn't go here. Sorry

File: drydock/templates/kustomized/tutor14/extensions/overrides.yml [hunk 2]
```diff
@@ -83,6 +84,7 @@ spec:
             requests:
               cpu: "{{ DRYDOCK_FORUM_REQUEST_CPU }}"
               memory: "{{ DRYDOCK_FORUM_REQUEST_MEMORY }}"
+{%- endif %}
```
Comment (mariajgrimaldi) [inline drydock/templates/kustomized/tutor14/extensions/overrides.yml:87]: Ignore this please. I'll remove it later

File: drydock/templates/kustomized/tutor13/extensions/debug/deployments.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          volumeMounts:
+            - mountPath: /openedx/edx-platform/lms/envs/tutor/
+              name: settings-lms
+            - mountPath: /openedx/edx-platform/cms/envs/tutor/
+              name: settings-cms
+            - mountPath: /openedx/config
+              name: config
+          resources:
+            requests:
+              memory: 2Gi
```
Comment (MoisesGSalas) [inline drydock/templates/kustomized/tutor13/extensions/debug/deployments.yml:45]: We probably shouldn't set a memory request for this kind of pod.

File: drydock/templates/kustomized/tutor13/extensions/debug/ingress.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-canary
```
Comment (MoisesGSalas) [inline drydock/templates/kustomized/tutor13/extensions/debug/ingress.yml:4]: `ingress-debug` seems like a better name.

File: drydock/templates/kustomized/tutor13/extensions/debug/ingress.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  namespace: {{ K8S_NAMESPACE }}
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    nginx.ingress.kubernetes.io/canary: "true"
+    nginx.ingress.kubernetes.io/canary-by-cookie: {{ OPENEDX_DEBUG_COOKIE }}
+    {%- if DRYDOCK_AUTO_TLS and not DRYDOCK_CUSTOM_CERTS%}
+    cert-manager.io/issuer: letsencrypt
+    {%- endif %}
+    argocd.argoproj.io/sync-wave: "5"
+spec:
```
Comment (MoisesGSalas) [inline drydock/templates/kustomized/tutor13/extensions/debug/ingress.yml:14]: Maybe we can add a default backend that points to caddy. IIRC the controller throws an error log if it doesn't find any service to fallback.

Comment (mariajgrimaldi) [reply drydock/templates/kustomized/tutor13/extensions/debug/ingress.yml:14]: I'll try that. Thanks!

**Review by MoisesGSalas — APPROVED** [review]
LGTM

**Pull request merged**